### PR TITLE
refactor: introduce election access route decorator

### LINKED ISSUE #380: Add decorator that transforms `election_id` param into `election` param, returns 404 if not found, returns 403 is not right permission
suggestion for later improvement: a decorator that transforms `election_id` param into `election` param, returns 404 if not found, returns 403 is not right permission.

_Originally posted by @benadida in https://github.com/votingworks/arlo/pull/376_

### PR DESCRIPTION
**Pull request by eventualbuddha**
**Description**

Closes #380
Refs #290

Decorating a route using `with_election_access` allows the route to be defined with an `election` parameter that will be be provided only if the election exists and the current user has access to it.

**Testing**

Existing tests should cover this okay, but to really test it I'd have to figure out how to test this more directly. I'm not sure what the right approach is there. Maybe create another Flask app per test that I then add test routes to?

**Progress**

This is a building block toward #290.

### CODE REVIEW

File: arlo_server/routes.py [hunk 1]
```diff
@@ -440,11 +438,15 @@ def update_jurisdictions_file(election_id=None):
...
-    ]
+    missing_fields = (
+        [
+            field
+            for field in [JURISDICTION_NAME, ADMIN_EMAIL]
+            if field not in jurisdictions_csv.fieldnames
+        ]
+        if jurisdictions_csv.fieldnames
+        else [JURISDICTION_NAME, ADMIN_EMAIL]
+    )
```
Comment (eventualbuddha) [inline arlo_server/routes.py]: I'm not sure why this change was required, but `mypy` all of a sudden started complaining:

```
arlo_server/routes.py:444: error: Unsupported right operand type for in ("Optional[Sequence[str]]")
```

I infer that `jurisdictions_csv.fieldnames` is `Optional[Sequence[str]]` and that `in` does not apply to `None`. But I don't understand why this is only now a problem since the type of this shouldn't have changed.

**Comment by jonahkagan** [comment]
Also refs #335

File: arlo_server/routes.py [hunk 2]
```diff
@@ -25,12 +25,13 @@
...
 from arlo_server import app
 from arlo_server.auth import (
+    UserType,
     clear_loggedin_user,
     get_loggedin_user,
     get_loggedin_user_record,
     require_audit_admin_for_organization,
     set_loggedin_user,
-    UserType,
+    with_election_access,
```
Comment (jonahkagan) [inline arlo_server/routes.py]: Could we add this to all the routes that have an `election_id` param and call `require_audit_admin_for_organization`?

File: arlo_server/auth.py [hunk 3]
```diff
@@ -55,3 +56,22 @@ def require_audit_admin_for_organization(organization_id: Optional[str]):
     raise Forbidden(
         description=f"{user.email} does not have access to organization {organization_id}"
     )
+
+
+def with_election_access(user_type: UserType):
```
Comment (jonahkagan) [inline arlo_server/auth.py]: Maybe a comment explaining what this does?

File: arlo_server/routes.py (hunk 1 above)
Comment (jonahkagan) [reply arlo_server/routes.py]: So weird! Here's an idea for something less verbose: 
```python
     missing_fields = [
        field
        for field in [JURISDICTION_NAME, ADMIN_EMAIL]
        if field not in (jurisdictions_csv.fieldnames or [])
    ]
```

**Comment by jonahkagan** [comment]
For testing, maybe just pick a really simple/basic endpoint and write all the tests for the logic of this decorator against that endpoint?

File: arlo_server/routes.py (hunk 2 above)
Comment (eventualbuddha) [reply arlo_server/routes.py]: Sure, forgot about other files!
